doc/api/url.md

```diff
@@ -2,136 +2,221 @@
 
     Stability: 2 - Stable
 
-This module has utilities for URL resolution and parsing.
-Call `require('url')` to use it.
+The `url` module provides utilities for URL resolution and parsing. It can be
+accessed using:
 
-## URL Parsing
+```js
+const url = require('url');
+```
+
+## The Components of a Parsed URL
+
+A URL is a structured string containing multiple meaningful components. When
+parsed, an object is returned containing properties for each of these
+components.
+
+The following details each of the components of a parsed URL. The example
+`'http://user:pass@host.com:8080/p/a/t/h?query=string#hash'` is used to
+illustrate each.
+
+### `href`
```

Comment:
Curious how those render in HTML. The GH markdown doesn't look pretty here, but I think there is no alternative. ff